### PR TITLE
Convert float velocity and steering to odom msg

### PR DESCRIPTION
**Pull request by averyjung39**

### CODE REVIEW

File: rr_arduino/rr_arduino.ino [hunk 1]
```diff
@@ -174,3 +181,29 @@ void get_battery_state(long current_time){
     prev_time = millis();
   }
 }
+
+void raw_to_odom(double vel, int str_angle) {
```
Comment (Oluwatoni) [inline rr_arduino/rr_arduino.ino]: the steering angle is passed in degrees when it is recieved, you can grab it there and use that

File: rr_arduino/rr_arduino.ino [hunk 2]
```diff
@@ -174,3 +181,29 @@ void get_battery_state(long current_time){
...
+    long current_time = millis();
+    long time_diff = current_time - prev_time;
+    // If the robot is at the origin, calculate the position using steering angle and the velocity
+    if (str_angle != 1500 && x == 0.0 && y == 0.0) {
+        // Mapping str_angle integer to angle in degrees
+        theta = (str_angle - 1000) / (2000 - 1000) * 60 - 30;
+        x = vel * cos(theta * M_PI / 180.0) * time_diff;
+        y = vel * sin(theta * M_PI / 180.0) * time_diff;
+
+        geometry_msgs::Quaternion odom_quat = tf::createQuaternionMsgFromYaw(theta);
```
Comment (Oluwatoni) [inline rr_arduino/rr_arduino.ino]: try compiling this

File: rr_arduino/rr_arduino.ino [hunk 3]
```diff
@@ -174,3 +181,29 @@ void get_battery_state(long current_time){
...
 }
+
+void raw_to_odom(double vel, int str_angle) {
+    long current_time = millis();
+    long time_diff = current_time - prev_time;
+    // If the robot is at the origin, calculate the position using steering angle and the velocity
+    if (str_angle != 1500 && x == 0.0 && y == 0.0) {
+        // Mapping str_angle integer to angle in degrees
+        theta = (str_angle - 1000) / (2000 - 1000) * 60 - 30;
+        x = vel * cos(theta * M_PI / 180.0) * time_diff;
```
Comment (Oluwatoni) [inline rr_arduino/rr_arduino.ino]: x is the integral so you would add the current value to what it was before

Comment (Oluwatoni) [inline rr_arduino/rr_arduino.ino]: your kinematic model is missing a few parameters, try following the equations in the blog you mentioned earlier. https://nabinsharma.wordpress.com/2014/01/02/kinematics-of-a-robot-bicycle-model/

File: rr_arduino/rr_arduino.ino (hunk 1 above)
Comment (Oluwatoni) [inline rr_arduino/rr_arduino.ino]: try to follow the function naming convention in https://google.github.io/styleguide/cppguide.html#Function_Names

File: rr_arduino/rr_arduino.ino [hunk 4]
```diff
@@ -174,3 +181,29 @@ void get_battery_state(long current_time){
     prev_time = millis();
   }
 }
+
+void raw_to_odom(double vel, int str_angle) {
+    long current_time = millis();
+    long time_diff = current_time - prev_time;
+    // If the robot is at the origin, calculate the position using steering angle and the velocity
+    if (str_angle != 1500 && x == 0.0 && y == 0.0) {
```
Comment (Oluwatoni) [inline rr_arduino/rr_arduino.ino]: try to keep the indentation at 2 spaces

File: rr_arduino/rr_arduino.ino (hunk 3 above)
Comment (averyjung39) [reply rr_arduino/rr_arduino.ino]: I just set x and y to the calculation assuming the code will enter this if statement when the car hasn't moved yet (i.e. when the car is in the origin). In the else statement I do calculations using the kinematics equations. I think this should be okay?

Comment (averyjung39) [reply rr_arduino/rr_arduino.ino]: I need the length of the car (L) to calculate the positions. I just put 0 as a placeholder.

File: rr_arduino/rr_arduino.ino [hunk 5]
```diff
@@ -174,3 +181,43 @@ void get_battery_state(long current_time){
     prev_time = millis();
   }
 }
+
+void RawToOdom(double vel, double str_angle) {
+  // FIX ME: Get length of the car
+  double L = 0;
```
Comment (Oluwatoni) [inline rr_arduino/rr_arduino.ino]: L = .335 meters

**Pull request closed without merging**